### PR TITLE
Feature/add password edit

### PR DESCRIPTION
**Pull request by janmaennig**
I have checked the extension in TYPO3v8. And I have added a functionality to edit the password if a user is logged in.

### CODE REVIEW

File: Classes/Controller/AuthenticationController.php [hunk 1]
```diff
@@ -180,8 +181,9 @@ public function showLogoutFormAction() {
     $user = $this->authenticationService->getAuthenticatedUser();
 
     if ($this->authenticationService->isUserAuthenticated()
-        && isset($formData['logintype'])
-        && $formData['logintype'] === LoginType::LOGIN) {
+      && isset($formData['logintype'])
```
Comment (mbrodala) [inline Classes/Controller/AuthenticationController.php]: Please do not reformat any unrelated code.

File: Classes/Controller/AuthenticationController.php [hunk 2]
```diff
@@ -194,6 +196,57 @@ public function showLogoutFormAction() {
...
+  /**
+   * Complete password edit
+   *
+   * @param string $password New password of the user
+   * @param string $passwordRepeat Confirmation of the new password
+   * @return void
+   * @validate $password NotEmpty
+   * @validate $passwordRepeat NotEmpty
+   */
+  public function completePasswordEditAction($password, $passwordRepeat) {
```
Comment (mbrodala) [inline Classes/Controller/AuthenticationController.php]: Make this `updatePasswordAction` instead please.

File: Resources/Private/Templates/Authentication/ShowLoginForm.html [hunk 3]
```diff
@@ -3,7 +3,7 @@
 <f:layout name="Default"/>
 <f:section name="main">
 
-<f:flashMessages renderMode="div"/>
+<f:flashMessages />
```
Comment (mbrodala) [inline Resources/Private/Templates/Authentication/ShowLoginForm.html]: This will cause a [different behavior in TYPO3 >= v7](https://docs.typo3.org/typo3cms/extensions/core/7.6/singlehtml/#breaking-63453-changed-rendering-of-flashmessagesviewhelper).

File: Resources/Private/Templates/Authentication/ShowPasswordEditForm.html [hunk 4]
```diff
@@ -0,0 +1,48 @@
+<f:layout name="Default"/>
+<f:section name="main">
+
+<f:flashMessages />
+
+<f:form.validationResults>
```
Comment (mbrodala) [inline Resources/Private/Templates/Authentication/ShowPasswordEditForm.html]: This should be put in a partial.

File: ext_emconf.php [hunk 5]
```diff
@@ -8,7 +8,7 @@
   'author_email' => 'mbrodala@pagemachine.de',
   'author_company' => 'PAGEmachine AG',
   'state' => 'stable',
-  'version' => '1.4.1',
+  'version' => '1.5.0-dev.1',
```
Comment (mbrodala) [inline ext_emconf.php]: Please don't change the version.

**Comment by janmaennig** [comment]
Thank you for your feedback. I am working on the changes promptly.
